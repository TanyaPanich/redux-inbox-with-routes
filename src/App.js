import React, {Component} from 'react'
import {connect} from 'react-redux'
import {bindActionCreators} from 'redux'
import './App.css'
import Messages from './components/Messages'
import Toolbar from './components/Toolbar'
import { initialize } from './actions/app.js'
import { withRouter } from 'react-router-dom'

class App extends Component {

  async componentDidMount() {
    this.props.initialize()
  }
  render() {
      return (<div className='App container'>
        <h1>Redux Inbox With Routes</h1>
        <Toolbar />
        <Messages />
      </div>)
  }
}

const mapStateToProps = (state) => {
  return {messagesState: state.messagesState}
}

const mapDispatchToProps = (dispatch) => bindActionCreators({
  initialize
}, dispatch)

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App))
