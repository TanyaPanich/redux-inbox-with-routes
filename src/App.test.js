/**
 * @jest-environment node
 */

import React from 'react'
import { MemoryRouter } from 'react-router'
import ReactDOM from 'react-dom'
import thunk from 'redux-thunk'
import { shallow } from 'enzyme'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import fetchMock from 'fetch-mock'
import renderer from 'react-test-renderer'
import App from './App'
const { JSDOM } = require('jsdom')

const jsdom = new JSDOM('<!doctype html><html><body></body></html>')
const { window } = jsdom
function copyProps(src, target) {
  const props = Object.getOwnPropertyNames(src)
    .filter(prop => typeof target[prop] === 'undefined')
    .reduce((result, prop) => ({
      ...result,
      [prop]: Object.getOwnPropertyDescriptor(src, prop),
    }), {})
  Object.defineProperties(target, props)
}

global.window = window
global.document = window.document
global.navigator = {
  userAgent: 'node.js',
}
copyProps(window, global)

const fakeMessages =
   [
     {message:
       {
         id: 11,
         subject: "Hey",
         starred: true,
         read: false,
         labels: [
           "dev",
           "personal"
         ]
        }
      }
    ]

 const middlewares = [ thunk ]
 const mockStore = configureMockStore(middlewares)
 const store = mockStore({
   messagesState:
   {
     messages: fakeMessages,
     allSelected: 'NONE',
     unreadCount: 0}
   }
 )

describe('App', () => {
  afterEach(() => {
    fetchMock.restore()
  })

  beforeEach(() => {
    fetchMock.get('/', fakeMessages)
  })

  it('renders without crashing', () => {
    shallow(<App />)
  })

  it('renders the same way every time', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>,
      document.getElementById('root')
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
