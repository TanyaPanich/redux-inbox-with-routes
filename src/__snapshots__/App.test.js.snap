// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the same way every time 1`] = `
<div
  className="App container"
>
  <h1>
    Redux Inbox With Routes
  </h1>
  <div
    className="row toolbar"
  >
    <div
      className="col-md-12"
    >
      <p
        className="pull-right"
      >
        <span
          className="badge badge"
        >
          0
        </span>
        unread messages
      </p>
      <a
        className="btn btn-danger"
      >
        <i
          className="fa fa-plus"
          onClick={[Function]}
        />
      </a>
      <button
        className="btn btn-default"
        onClick={[Function]}
      >
        <i
          className="fa fa-square-o"
        />
      </button>
      <button
        className="btn btn-default"
        onClick={[Function]}
      >
        Mark As Read
      </button>
      <button
        className="btn btn-default"
        onClick={[Function]}
      >
        Mark As Unread
      </button>
      <select
        className="form-control label-select"
        onChange={[Function]}
      >
        <option
          value={false}
        >
          Apply label
        </option>
        <option
          value="dev"
        >
          dev
        </option>
        <option
          value="personal"
        >
          personal
        </option>
        <option
          value="gschool"
        >
          gschool
        </option>
      </select>
      <select
        className="form-control label-select"
        onChange={[Function]}
      >
        <option>
          Remove label
        </option>
        <option
          value="dev"
        >
          dev
        </option>
        <option
          value="personal"
        >
          personal
        </option>
        <option
          value="gschool"
        >
          gschool
        </option>
      </select>
      <button
        className="btn btn-default"
        onClick={[Function]}
      >
        <i
          className="fa fa-trash-o"
        />
      </button>
    </div>
  </div>
  <div
    className="Messages"
  >
    <div>
      <div
        className="row message unread "
      >
        <div
          className="col-xs-1"
        >
          <div
            className="row"
          >
            <div
              className="col-xs-2"
            >
              <input
                checked=""
                onChange={[Function]}
                type="checkbox"
              />
            </div>
            <div
              className="col-xs-2 star fa fa-star"
              onClick={[Function]}
            >
              <i />
            </div>
          </div>
        </div>
        <div
          className="col-xs-11"
        >
          <span
            className="label label-warning"
          >
            dev
          </span>
          <span
            className="label label-warning"
          >
            personal
          </span>
          <a
            onClick={[Function]}
          >
            Hey
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
